osf/models/subject.py

```diff
@@ -1,9 +1,14 @@
 # -*- coding: utf-8 -*-
 from django.db import models
+from django.core.exceptions import ValidationError
 
 from website.util import api_v2_url
 
 from osf.models.base import BaseModel, ObjectIDMixin
+from osf.models.preprint_provider import PreprintProvider
+from osf.models.preprint_service import PreprintService
+from osf.models.validators import validate_subject_hierarchy_length, validate_subject_provider_mapping
+from osf.utils.caching import cached_property
```

Comment:
Use  django's cached_property. You only need osf.utils.caching.cached_property if you need to invalidate the cached property when it returns `None` or you need a setter.